src/Connector/BaseApiResourceConnector.php

```diff
@@ -87,13 +87,38 @@ public function getResource()
     }
 
     /**
+     * @return mixed
+     */
+    public function resourceExists()
+    {
+        return $this->resource->exists();
+    }
+
+    /**
+     * @return array
+     */
+    public function getDataForUpdate()
+    {
+        return $this->resource->getChanges();
+
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getDataForPost()
+    {
+        return $this->resource->getMutableData();
+    }
+
+    /**
      * @return null|string
      * @throws \Exception
      */
     public function getEndpoint()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this function "getEndpoint" is 5 which is greater than 4 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=php%3AS1541)